### ISSUE TITLE
Upgrade your plan 

### ISSUE THREAD

**Issue by hafsazulf17** [socioboard/Socioboard-5.0#361]
On login to dashboard It says :
Please upgrade your account 
How can we fix it from the backend currently the values are setup as:
`update user_activations set user_plan = 7;`

created at : 2022-03-19 15:05:55
account_expire_date : 2022-04-02 15:05:55

**Comment by sersart** [comment]
Hi
you have two way:
1-Install amember and add all products, add amember settings info to development.json file
2-Comment or remove amember connection from User microservice

**Comment by smokinjo** [comment]
So, amember is bascically an integral part of SocioBoard.  It is written into SocioBoard.

Are there plans to add othr non-amember options to SocioBoard?

amember seems to be a solution geared towards those wanting to recreate a sociobpard to sell to others.

But, the way it is presented, it is for people and teams trying to use socioboard for their own campaigns an dsmall teams.

But, so far, there is not a way to manage small numbers of people.  Amember is a great tool when recreating socioboard for lots  of users.  But, for a small team, there are no member ship levels.  Just the team.  Amember is too powerful of a tool to assign access to a handful of people in 1-2 teams.

Are there other options available, or simple tools to be able to create users and teams?

**Comment by sersart** [comment]
You have to remove all hard coded amember connections from app.

**Comment by hafsazulf17** [comment]
> You have to remove all hard coded amember connections from app.

@sersart  where to remove all hard coded amember connections from app , there is guide file in repository?? Is amember a seperate 3rd party software integrated with API or what is it? Please help me to understand it?

**Comment by sersart** [comment]
Hi,
1- Check async register(req, res, next)  function in file 
Socioboard-5.0/socioboard-api/User/core/unauthorized/unauthorized.service.js
this function adding member to amember

2- async login(req, res, next) function in same file

Let me know if you need help

**Comment by hafsazulf17** [comment]
how can we install amember ?  You have mentioned : 1-Install amember and add all products, add amember settings info to development.json file

**Comment by sersart** [comment]
https://docs.amember.com/Installation/Installation/

and add info to  Socioboard-5.0/socioboard-api/User/config/development.json
  "aMember": {
    "key": "<< aMember Key >>",
    "domain": "https://<< DOMAIN >>/amember/api"
  },

**Comment by arrianda** [comment]
Can you edit the database directly to edit column "account_expire_date" far more in the future?
Or you can edit default.json config at user microservice and set value for "user_base_plan_expiry_days" for longer days. This only works for new user.

**Comment by sersart** [comment]
Check this file
/socioboard-api/Common/Mappings/amember.users.js 
function:
  getUserPlanDetail(userId = 0, userName = '') {

            // Add subscription to the user.
            this.setPlanDetailsToUser(userId, planId, expiryDate)
              .then(response => {
                resolve(response);
              })
              .catch(error => {
                logger.error(
                  `Error amember Inner getUserPlanDetail: ${error?.message}`
                );
                resolve(true);
              });

 
Let me know if you need help

**Comment by hafsazulf17** [comment]
@sersart 
Is a member free and can we download it by signing up also which folder do we have to install a member ?
Thanks

**Comment by vaughngx4** [comment]
> Can you edit the database directly to edit column "account_expire_date" far more in the future?
> Or you can edit default.json config at user microservice and set value for "user_base_plan_expiry_days" for longer days. This only works for new user.

Yes you can. Run this command in mysql: `update user_activations set expiry_date = "2999-12-30";`

**Comment by vaughngx4** [comment]
@hafsazulf17 @sersart I've added an admin panel and an option to disable aMember. See the `Docker.README.md` and `Configuration.README.md`. Use my fork until PR #384 is merged.

**Comment by sureshbabu1995** [comment]
Hello Guys,

It appears that you have received the information you requested. I suggest we close this inactive thread for the time being. If you have any further concerns, you can create a new ticket.
